### ISSUE TITLE
camelCase util function does not return what it states for the sample input

### ISSUE THREAD

**Issue by bwbroersma** [roots/sage#2355]
* [x] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
* [x] This request isn't a duplicate of an existing issue
* [x] I've read the [docs](https://roots.io/documentation/) and followed them (if applicable)
* [x] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

## Description

See the current code:
https://github.com/roots/sage/blob/bd38502616188be973f60b75717eea95c1fc69d2/resources/assets/scripts/util/camelCase.js#L2-L9
The camelCase util function does not return what it states for the sample input (except being the "the most terrible camelizer"), is this by design? (if so, maybe add a comment about that)

## Steps to reproduce

**Actual behavior:**
input `"CAMeL_CaSEiS-harD"` => output `"cAMeLCaSEiSHarD"`

**Expected behavior:**
if `.toLowerCase()` is added in the `map()` the result will be `"camelCaseisHard"`.
Note that the change will change the correctly camelCase input `"camelCaseIsHard"` => output `"camelcaseishard"`, so it will convert the uppercase characters to lowercase.

**Comment by QWp6t** [comment]
> the most terrible camelizer on the internet

Sounds like it's working as advertised 😏

**Comment by bwbroersma** [comment]
> Sounds like it's working as advertised smirk

I'm actually not sure what it should do, it's not really clear, especially since the output example does not match the input example. I was looking into the javascript of one of my colleagues and was amazed by this function. Since 'a fix' would change the behavior, I'm asking this question.
If it is really intended to be terrible, maybe add 'this will produce cAMeLCaSEiSHarD' to the example input.

**Comment by knowler** [comment]
@bwbroersma The function wasn’t meant to be used generally outside of [Sage 9’s DOM router](https://github.com/roots/sage/blob/master/resources/assets/scripts/util/Router.js) since it only meets the needs of the router (camelcasing CSS class names iirc). [The new version of the router](https://github.com/roots/js-dom-router) doesn’t use the camelcasing anymore and Sage 10 doesn’t even include any JS DOM router since it recommends code splitting instead.

In other words, use something like Lodash instead.
